### PR TITLE
New graphql server

### PR DESCRIPTION
**Pull request by mrjackv**

### CODE REVIEW

File: python/revng/cli/daemon.py [hunk 1]
```diff
@@ -13,37 +14,56 @@ class DaemonCommand(Command):
...
 
-    def register_arguments(self, parser):
-        parser.add_argument("-p", "--port", type=str)
+    def register_arguments(self, parser: ArgumentParser):
+        parser.add_argument("-p", "--port", type=str, default="8000", help="Port to use")
+        parser.add_argument(
+            "--export",
+            action="store_true",
+            help="Don't start the server, only print the variables that would be used to start it",
+        )
```
Comment (aleclearmind) [inline python/revng/cli/daemon.py:23]: Do we still need this now that we have the production server in-house?

**Comment by aleclearmind** [comment]
Please justify in the commit message the reasons for which we're switching away from flask.

File: python/revng/cli/daemon.py [hunk 2]
```diff
@@ -13,37 +14,56 @@ class DaemonCommand(Command):
...
+        parser.add_argument("--prod", action="store_true", help="Production mode")
```
[the remark this answers is not in the document]
Comment (mrjackv) [reply python/revng/cli/daemon.py]: 1. Fixed
2. Reworded the help message to clarify

File: python/revng/cli/daemon.py (hunk 1 above)
Comment (mrjackv) [reply python/revng/cli/daemon.py]: I dropped `--export`
However I added an option `--hypercorn-args` to allow some flexibility while running the server (e.g. enable hot reload), which seems the real reason behind the need for `--export`

**Comment by mrjackv** [comment]
> Please justify in the commit message the reasons for which we're switching away from flask.

Reworded the main commit, should be more explicative.

**Comment by aleclearmind** [comment]
Please comment on why we're switching to hypercorn in the commit. Then, I think we're good to go.

**Comment by aleclearmind** [comment]
Merged in f42fd7f. Thanks!

**Pull request closed without merging**